packages/schema/lib/schemas/ThrottleObjectSchema.js

```diff
@@ -19,11 +19,17 @@ module.exports = makeSchema({
         'The maximum number of invocations for an action, allowed within the timeframe window.',
       type: 'integer',
     },
+    key: {
+      description:
+        'The key to throttle with in combination with the scope. This should be unique to the operation. While actions of different integrations with the same key and scope will never share the same limit, actions of the same integration with the same key and scope will do when "action" is not in the scope. User data provided for the input fields can be used in the key with the use of the curly braces referencing. For example, to access the user data provided for the input field "test_field", use `{{bundle.inputData.test_field}}`. Note that a required input field should be referenced to get user data always.',
```

Comment:
Can the user _not_ provide a key?

Reply:
What happens in the scenario where a user uses the same key in two actions, but has different values?

For example:
```
update_row: {
  ...
  throttle: {
    key: 'row_level_action',
    limit: 50,
    window: 60,
    scope: ['account']
  }
}

delete_row: {
  ...
  throttle: {
    key: 'row_level_action',
    limit: 1000,
    window: 60,
    scope: ['account']
  }    
}
```

Reply:
Good question!

Looking at https://gitlab.com/zapier/zapier/-/blob/fb1bdfa5dbb019ddac4a599e2d303c13cd3a712d/web/backend/zapier/useful/throttle_tools.py#L133-135

- `window` is mixed into the `key`, so if those don't match they won't affect each other. That should be called out in the above description (comment).
- `limit` is applied per action, so if in your example `window` would be identical, then within that window `update_row` would throttle at 50, but `delete_row` would happily continue up to 1000. 🤔

Reply:
I see - I think there's 3 ways to go about this then:
1. Remove `window` from key as Fokke suggested
2. Add `limit` to the key as Kola suggested
3. Add a validation assertion to this schema: if a throttle key is defined, the limit and key must be equal and when the user runs `zapier validate` they will see a validation error.

1 and 2 carry the risk of unintentional behaviour, where a user may report that they thought their action will throttle at X/Y but is actually doing A/B. But we can be very loud about documenting this behaviour to try and mitigate this.

While 3 carries more work and effort, I think this will lead to implementing the original intent behind adding a throttle key. 

I'm fine with any - if 1/2, then I suggest adding a clear message and maybe a snippet of the above example
@FokkeZB @kola-er wdyt?

Reply:
Aside from supporting the new `key`, the main goal is to support the sharing of the same limit across multiple actions without causing unintended behavior. The scenario you painted was helpful to see the possibility of unintended behavior. However, I think validating would be restrictive and could be limiting the devs for some other intended behaviors. As for the option of removing the `window` from being mixed into the final key, it would still result to both limits of `50` and `1000` being used but not both in the same timeframe.

Adding the `limit` to the final key, in addition to the `window` that is already mixed into it, we can have a description with less confusion like this;

```
The final key used for the throttling is formed as a combination of all the configurations; key, window, limit, and scope. To share a limit across multiple actions, each should have the same configuration set without "action" in the scope.
```

This way, devs know exactly how they need to set the configuration to share a limit across actions without unintended behavior.

Reply:
Btw, this PR can be merged and the case of unintended behavior can be treated in a separate release to not delay the [15.6.0 release](https://github.com/zapier/zapier-platform/pull/758).

Reply:
That seems reasonable, thanks for discussing this!

Reply:
I am happy to move forward with adding the `limit` value to the final key

Reply:
Okay, will include it in the backend code and update the docs here too. Thanks @standielpls

Reply:
@standielpls Updated the description to note that all the configurations are combined to form the final key in the last commit.

Reply:
This makes sense. Thanks all.

Also see https://gitlab.com/zapier/zapier/-/merge_requests/56351#note_1830368072